pkg/generate/app/cmd/newapp.go

```diff
@@ -222,26 +224,56 @@ func (c *AppConfig) SetOpenShiftClient(osclient client.Interface, OriginNamespac
 // AddArguments converts command line arguments into the appropriate bucket based on what they look like
 func (c *AppConfig) AddArguments(args []string) []string {
 	unknown := []string{}
+	c.ArgumentClassificationErrors = map[string]error{}
 	for _, s := range args {
-		switch {
```

Comment:
If you're ditching the switch statement, can you refactor into a separate function which then allows you to avoid all the nested ifs? i.e.:
```
if foo {
    // something
    return
}

if bar {
    // something else
    return
}
```

rather than

```
if foo {
    // something
} else {
    if bar {
        // something else
    } else {
        if baz ...
    }
}